ReleaseProcessScript.New.Tests/CommandLineGitClientTests.cs

```diff
@@ -0,0 +1,310 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using NUnit.Framework;
+using Optional;
+using Optional.Unsafe;
+
+namespace ReleaseProcessScript.New.Tests;
+
+[TestFixture]
+internal class CommandLineGitClientTests
+{
+  [SetUp]
+  public void Setup ()
+  {
+    _prevDir = Environment.CurrentDirectory;
+    var temp = Path.GetTempPath();
+
+    var guid = Guid.NewGuid();
+    var path = Path.Combine(temp, guid.ToString());
+    _remote = Directory.CreateDirectory(path).FullName;
+
+    Environment.CurrentDirectory = _remote;
+    _remoteName = "origin";
+
+    ExecuteGitCommand("--bare init");
+
+    guid = Guid.NewGuid();
+
+    path = Path.Combine(temp, guid.ToString());
+    _dir = Directory.CreateDirectory(path).FullName;
+
+    Environment.CurrentDirectory = _dir;
+
+    ExecuteGitCommand("init");
+    ExecuteGitCommand($"remote add {_remoteName} " + _remote);
+    ExecuteGitCommand("commit -m \"Initial Commit\" --allow-empty");
+  }
+
+  [TearDown]
+  public void TearDown ()
+  {
+    Environment.CurrentDirectory = _prevDir;
+    DeleteDirectory(_dir);
+    DeleteDirectory(_remote);
+  }
+
+  private string _prevDir;
+  private string _dir;
+  private string _remote;
+  private string _remoteName;
+
+  [Test]
+  public void GetBranches_WithTwoBranches_ReturnsBranches ()
+  {
+    ExecuteGitCommand("branch develop");
+    var client = new CommandLineGitClient(_dir);
+
+    var branches = client.GetBranches();
+
+    Assert.That(branches, Does.Contain("master"));
+    Assert.That(branches, Does.Contain("develop"));
+  }
+
+  [Test]
+  public void GetCurrentBranchName_WithOneBranch_ReturnsBranchName ()
+  {
+    var client = new CommandLineGitClient(_dir);
+
+    var branchName = client.GetCurrentBranchName();
+
+    Assert.That(branchName.ValueOrDefault(), Is.EqualTo("master"));
+  }
+
+  [Test]
+  public void GetCurrentBranchName_WithTwoBranches_ReturnsBranchName ()
+  {
+    var client = new CommandLineGitClient(_dir);
+    ExecuteGitCommand("branch develop");
+
+    var branchName = client.GetCurrentBranchName();
+
+    Assert.That(branchName.ValueOrDefault(), Is.EqualTo("master"));
+  }
+
+  [Test]
+  public void BranchExists_WithOneBranch_ReturnsTrue ()
+  {
+    var client = new CommandLineGitClient(_dir);
+
+    var outcome = client.BranchExists("master");
+
+    Assert.That(outcome, Is.True);
+  }
+
+  [Test]
+  public void RemoteBranchExists_BranchExists_ReturnsTrue ()
+  {
+    var client = new CommandLineGitClient(_dir);
+    ExecuteGitCommand("checkout -b newBranch --quiet");
```

Comment:
All `--quiet` or `-q` flags can be removed in this file, as this is no longer improving test output readability as it did in powershell.